src/Centreon/Domain/Common/Exception/FactoryException.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+/*
+ * Copyright 2005 - 2021 Centreon (https://www.centreon.com/)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+declare(strict_types=1);
+
+namespace Centreon\Domain\Common\Exception;
+
+/**
+ * This class is designed to contain all entity factory exceptions.
+ *
+ * @package Centreon\Domain\Common\Exception
+ */
+class FactoryException extends \Exception
+{
+    /**
+     * @param string $className
+     * @return FactoryException
+     */
+    public static function mandatoryDataNotFound(string $className): self
+    {
+        return new self(
+            sprintf(_('Mandatory data to create the entity \'%s\' is missing'), $className)
```

Comment:
you should add translation in `.po` file

Reply:
it is:
msgid "Mandatory data to create the entity '{%s}' is missing"
msgstr "Les données obligatoires pour créer l'entité '{%s}' sont manquantes"